paasta_tools/spark_tools.py

```diff
@@ -1,13 +1,202 @@
+
+
+class SparkEnvironmentConfig(TypedDict):
+    """
+    There's no set schema for the spark configuration from DEFAULT_SPARK_RUN_CONFIG,
+    but at the time of writing, this file looks like (with the addition of a `prod` key):
+    environments:
+        dev:
+            account_id: 'SOME_ACCOUNT_ID'
+            default_event_log_dir: s3a://SOME_BUCKET  # currently the only thing paasta reads
+            history_server: http://SOME_URL/
+    """
+
+    environments: Dict[str, SparkConfig]
+
+
+SparkEventLogConfiguration = TypedDict(
+    "SparkEventLogConfiguration",
+    {"spark.eventLog.enabled": str, "spark.eventLog.dir": str},
+    total=False,
+)
+
+
+@lru_cache(maxsize=1)
+def get_default_spark_configuration() -> Optional[SparkEnvironmentConfig]:
+    """
+    Read the globally distributed Spark configuration file and return the contents as a dictionary.
+
+    At the time this comment was written, the only bit of information that we care about from this file
+    is the default event log location on S3. See the TypedDict of the retval for the bits that we care about
+    """
+    try:
+        with open(DEFAULT_SPARK_RUN_CONFIG, mode="r") as f:
+            return yaml.safe_load(f.read())
+    except OSError:
+        log.error(
+            f"Unable to open {DEFAULT_SPARK_RUN_CONFIG} and get default configuration values!"
+        )
+    except yaml.YAMLError:
+        log.error(
+            f"Unable to parse {DEFAULT_SPARK_RUN_CONFIG} and get default configuration values!"
+        )
+
+    return None
+
 
 def get_webui_url(port: str) -> str:
     return f"http://{socket.getfqdn()}:{port}"
 
 
+def setup_event_log_configuration(spark_args: Dict[str, str]) -> Dict[str, str]:
+    """
+    Adjusts user settings to provide a default event log storage path if event logging is
+    enabled but not configured.
+
+    If event logging is not enabled or is fully configured, this function will functionally noop.
+    """
+    # don't enable event logging if explicitly disabled
+    if spark_args.get("spark.eventLog.enabled", "true") != "true":
+        # Note: we provide an empty dict as our return value as the expected
+        # usage of this function is something like CONF.update(setup_event_log_configuration(...))
+        # in in this case, we don't want to update the existing config
+        return {}
+
+    # user set an explicit event log location - there's nothing else for us to
+    # do here
+    if spark_args.get("spark.eventLog.dir") is not None:
+        # so, same as above, we return an empty dict so that there are no updates
+        return {}
+
+    default_spark_conf = get_default_spark_configuration()
+    if default_spark_conf is None:
+        log.error(
+            "Unable to access default Spark configuration, event log will be disabled"
+        )
+        # Note: we don't return an empty dict here since we want to make sure that our
+        # called will overwrite the enabled option with our return
```

Comment:
err, I made a typo here :p

this should have read `Note: we don't return an empty dict here since we want to make sure that our caller will overwrite the enabled option with our return`

this'll make a little more sense with the next PR in the series, but this function is expected to be called like `SOMEDICT.update(setup_event_log_configuration(USER_SPARK_ARGS_HERE))` - so we want to return a dict with the kv pair that disables this spark feature so that the dict that we're `.update`ing actually gets modified

Reply:
i tried clarifying this a bit more - thoughts?

Reply:
Makes sense, thanks for the extra information!